src/int.cr

```diff
@@ -155,14 +155,18 @@ struct Int
   #
   # See `Int#/` for more details.
   def %(other : Int)
-    if other == 0
-      raise DivisionByZeroError.new
-    elsif (self < 0) == (other < 0)
-      self.unsafe_mod(other)
-    else
-      me = self.unsafe_mod(other)
-      me == 0 ? me : me + other
-    end
+    {% begin %}
+      if other == 0
+        raise DivisionByZeroError.new
+      elsif self < 0 && self == {{@type}}::MIN && other == -1
```

Comment:
What happens for `other == -2` ?

Reply:
It seems to work fine. It gives 0.

Reply:
That should be a spec

Reply:
But the bug mentioned `-1` specifically. I'm not sure we need to add specs for -2, -3 or possibly any number that might have a bug.

Reply:
IIUC it stresses different path in the code, so it might be worth having a test coverage for it?